lucene/core/src/java/org/apache/lucene/search/TopDocsCollector.java

```diff
@@ -136,12 +136,9 @@ public TopDocs topDocs(int start, int howMany) {
     // pq.size() or totalHits.
     int size = topDocsSize();
 
-    // Don't bother to throw an exception, just return an empty TopDocs in case
-    // the parameters are invalid or out of range.
-    // TODO: shouldn't we throw IAE if apps give bad params here so they dont
-    // have sneaky silent bugs?
+
     if (start < 0 || start >= size || howMany <= 0) {
-      return newTopDocs(null, start);
+      throw new IllegalArgumentException("Illegal arguments when requesting top docs");
```

Comment:
can you add the value of start and howMany to the error message and what the expectation is for their values?